inject/src/main/java/io/micronaut/context/env/DefaultEnvironment.java

```diff
@@ -99,6 +99,7 @@
     private static final String DO_SYS_VENDOR_FILE = "/sys/devices/virtual/dmi/id/sys_vendor";
     private static final Boolean DEDUCE_ENVIRONMENT_DEFAULT = true;
     private static final List<String> DEFAULT_CONFIG_LOCATIONS = Arrays.asList("classpath:/", "file:config/");
+    private static final String DEFAULT_ENVIRONMENT_NAME = "default";
```

Comment:
Can you explain this? Why would the constant property source be called default over application?

Reply:
`application` is not the name of the environment, but the name of the _resource_ (that is to say it corresponds to the base name in `application-[env].yml`). There is special casing for the "default" case where the `-[env]` part is not used when searching for resources. So the complete pattern is `[resource name]-[env name]`, except for the default environment.

Reply:
There is no default environment, unless the user specifies one. That is why this doesn't make sense to me

Reply:
The "default" application configuration is always loaded. However in the context of AOT, we need to make a difference between "the user didn't configure anything" and "the user explicitly chose to limit AOT to the default environment, ignoring `-dev`, `-aws` and all). For this we need a way to denote the "default" environment name, which is, the one with no name.

Reply:
The default environment is always loaded though so perhaps this is not necessary and the logic should instead always load the default env?

Reply:
I removed the special case for `default` and rebased onto `3.3.x` instead.